Source/diffusion/Castro_diffusion.cpp

```diff
@@ -173,6 +173,99 @@ Castro::getTempDiffusionTerm (Real time, MultiFab& state_in, MultiFab& TempDiffT
        FillPatch(getLevel(level-1),CrseTemp,1,time,State_Type,UTEMP,1);
    }
 
-   diffusion->applyop(level, Temperature, CrseTemp, TempDiffTerm, coeffs);
+   if (diffuse_use_amrex_mlmg) {
+       // Evaluates ∇ ⋅(k_th ∇T) using AMReX
+       diffusion->applyop(level, Temperature, CrseTemp, TempDiffTerm, coeffs);
+   } else {
+       // Evaluates ∇ ⋅(k_th ∇T) without using AMReX
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+       for (MFIter mfi(Temperature, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+           {
+               const auto dx = geom.CellSizeArray();
+               const auto dxinv = geom.InvCellSizeArray();
+               const auto problo = geom.ProbLoArray();
+               const int coord = geom.Coord();
+
+               const Box& bx = mfi.tilebox();
+               Array4<Real const> const& Temp_array = Temperature.array(mfi);
+               Array4<Real> const & TempDiff_array = TempDiffTerm.array(mfi);
+
+               for (int idir = 0; idir < AMREX_SPACEDIM; ++idir) {
+
+                   // note coeffs is edged based k_th
+                   Array4<Real const> const edge_coeff_arr = (*coeffs[idir]).array(mfi);
```

Comment:
it might be better to just grab all of the edge coeffs into an array of Array4 and then put the loop over directions inside the GPU lambda -- that way there are fewer GPU kernel launches (one 1 instead of 1 per dimension)